Content.Server/Clothing/Components/RestrictedClothingComponent.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Content.Server.Speech.Components;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Server.Clothing.Components;
+
+/// <summary>
+///     Does bad things when someone tries to wear something they shouldn't
+/// </summary>
+[RegisterComponent]
+public sealed class RestrictedClothingComponent : Component
+{
+    /// <summary>
+    ///     List of permissions required to be allowed to wear the item
+    /// </summary>
+    [DataField("permissions")] [ViewVariables(VVAccess.ReadWrite)]
+    public HashSet<string> Permissions = new();
+
+    /// <summary>
+    ///     Uid that won't be affected
+    /// </summary>
+    [DataField("whitelistedUid")] [ViewVariables(VVAccess.ReadWrite)]
+    public int WhitelistedUid = 0;
```

Comment:
shouldnt this be an `EntityUid` ?

Reply:
Hacky code should be documented as such but in this case it should still be updated.

Reply:
Permissions duplicates whitelist so requireswhitelist still needs purging.